File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res(2);
        unordered_map<int, int> mp;
        for(int i=0; i<nums.size(); i++){
            if(mp.find(target - nums[i]) != mp.end()){
                res[1] = i;
                res[0] = mp[target - nums[i]];
                return res;
            }
            mp[nums[i]] = i;
        }
        return res;
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        if(n <= 2)
            return new int[] {1,0};
        HashMap<Integer, Integer> mp = new HashMap<>();
        int[] res= new int[2];
        for(int i=0; i<n; i++){
            int req = target - nums[i];
            if(mp.get(req) != null){
                res[1] = i;
                res[0] = mp.get(req);
                return res;
            }else{
                mp.put(nums[i], i);
            }
        }
        return res;
    }
}